src/common/selectors/selectorUtils.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Messages } from '@salesforce/core';
+import { Connection, QueryResult, Record } from 'jsforce';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-devops-center', 'commonErrors');
+
+export async function runSafeQuery<T extends Record>(con: Connection, queryStr: string): Promise<QueryResult<T>> {
+  // Parameters verification
+  if (!con) {
+    throw messages.createError('error.connection-required');
+  }
+  if (typeof queryStr !== 'string' || queryStr.trim().length === 0) {
+    throw messages.createError('error.query-string-required');
+  }
+
+  // Run the query and verify for results
+  try {
+    const result: QueryResult<T> = await con.query(queryStr);
+    if (result.totalSize > 0 && result.records) {
```

Comment:
In our Heroku app we also check 'result.done'.  Is that needed here?

Also, do we have any any queries (or will we) that could return an empty list?

Reply:
Yes, fetchAsyncOperationResult() when we filter by the Id provided by the user in resume.

Reply:
If we get here a result.done = false, do we want to do a querymore() ???

Reply:
I still think this should support an optional parameter to allow it to return a null result (when totalSize === 0).   Then we can harden the logic in abstractReportOnPromote